packages/template-retail-react-app/app/commerce-api/auth.js

```diff
@@ -79,23 +85,68 @@ class Auth {
     }
 
     get authToken() {
-        return this._authToken
+        return this._storage.get(tokenStorageKey)
+    }
+
+    set authToken(token) {
+        this._storage.set(tokenStorageKey, token)
+    }
+
+    get userType() {
+        return this._storage.get(refreshTokenRegisteredStorageKey)
+            ? Auth.USER_TYPE.REGISTERED
+            : Auth.USER_TYPE.GUEST
     }
 
     get refreshToken() {
-        return this._refreshToken
+        const storageKey =
+            this.userType === Auth.USER_TYPE.REGISTERED
+                ? refreshTokenRegisteredStorageKey
+                : refreshTokenGuestStorageKey
+        return this._storage.get(storageKey)
     }
 
     get usid() {
-        return this._usid
+        return this._storage.get(usidStorageKey)
+    }
+
+    set usid(usid) {
+        this._storage.set(usidStorageKey, usid)
     }
 
     get encUserId() {
-        return this._encUserId
+        return this._storage.get(encUserIdStorageKey)
+    }
+
+    set encUserId(encUserId) {
+        this._storage.set(encUserIdStorageKey, encUserId)
     }
 
     get oid() {
-        return this._oid
+        return this._storage.get(oidStorageKey)
+    }
+
+    set oid(oid) {
+        this._storage.set(oidStorageKey, oid)
+    }
+
+    /**
+     * Save refresh token in designated storage.
+     *
+     * @param {string} token The refresh token.
+     * @param {USER_TYPE} type Type of the user.
+     */
+    _saveRefreshToken(token, type) {
+        if (type === Auth.USER_TYPE.REGISTERED) {
+            this._storage.set(refreshTokenRegisteredStorageKey, token, {
+                expires: REFRESH_TOKEN_COOKIE_AGE
+            })
+            this._storage.delete(refreshTokenGuestStorageKey)
+            return
+        }
+
+        this._storage.set(refreshTokenGuestStorageKey, token, {expires: REFRESH_TOKEN_COOKIE_AGE})
+        this._storage.delete(refreshTokenRegisteredStorageKey)
```

Comment:
I feel like the clearing of the "other" key value from storage shouldn't exist in a function called "saveRefreshToken. Maybe they can be cleared out before this function is called? This would leave you with a cleaner implementation of this function that looked something like this: 

```suggestion
        const key = type === Auth.USER_TYPE.REGISTERED ? 
            refreshTokenRegisteredStorageKey :
            refreshTokenGuestStorageKey
            
        this._storage.set(key, token, {expires: REFRESH_TOKEN_COOKIE_AGE})
```

Another idea is to name it updateRefreshToken.. then I would be ok with the clearing happening inside.